### PR TITLE
Add: Documentation to source

### PR DESCRIPTION
**Pull request by oliverjiu**
## Description

- Added documentation to source.

### CODE REVIEW

File: src/pipeline/auxiliary/number.ts [hunk 1]
```diff
@@ -1,109 +1,116 @@
...
-		}
-	).text; // Our accumulator is an object containing text, thus we assign a text property
-}
+/*
+	File: src/pipeline/auxiliary/number.ts
+	cpuabuse.com
+*/
+
+/**
+ * Adds numbers to the begining of the line
```
Comment (yanocv) [inline src/pipeline/auxiliary/number.ts]: Missing a dot.

File: src/pipeline/auxiliary/number.ts [hunk 2]
```diff
@@ -1,109 +1,116 @@
...
+ */
+
+const emptyString: string = "";
+const space: string = " ";
+const tabulationSpace: string = " ";
+let treatCarriageReturnAsNotEmpty: boolean = false;
+
+/**
+ * Determinate if the function is empty.
+ * @param text Text to check.
```
Comment (yanocv) [inline src/pipeline/auxiliary/number.ts]: No need for dot.

File: src/pipeline/auxiliary/number.ts [hunk 3]
```diff
@@ -1,109 +1,116 @@
...
+		text.length === 0 ||
+		(treatCarriageReturnAsNotEmpty // Account for Windows line endings
+			? false
+			: text === "\r")
+	);
+}
+
+/**
+ * Numbers the line.
+ * @param text Text to check.
```
Comment (yanocv) [inline src/pipeline/auxiliary/number.ts]: No dots necessary.

File: src/pipeline/auxiliary/number.ts [hunk 4]
```diff
@@ -1,109 +1,116 @@
...
+ * @param countBlank Text to check.
```
Comment (yanocv) [inline src/pipeline/auxiliary/number.ts]: No need for dot.

**Comment by winterhotlatte** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 25
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/cpuabuse/cross-cat/pullRequest?prid=5215125&bid=17100350)

File: src/pipeline/show-nonprinting.ts [hunk 5]
```diff
@@ -1,70 +1,111 @@
...
-		) {
-			array[index] = `^${String.fromCharCode(characterCode + firstAlphaChar)}`; // eslint-disable-line no-param-reassign
-		}
-	});
-	return resultArray.join(emptyString);
-}
+/*
+	File: src/pipeline/show-nonprinting.ts
+	cpuabuse.com
+
```
Comment (yanocv) [inline src/pipeline/show-nonprinting.ts]: No need for the extra line.

**Review by yanocv — CHANGES_REQUESTED** [review]
Several changes needed.

File: src/pipeline/auxiliary/number.ts (hunk 2 above)
Comment (oliverjiu) [reply src/pipeline/auxiliary/number.ts]: Solved.

File: src/pipeline/auxiliary/number.ts (hunk 3 above)
Comment (oliverjiu) [reply src/pipeline/auxiliary/number.ts]: Solved.

File: src/pipeline/auxiliary/number.ts (hunk 4 above)
Comment (oliverjiu) [reply src/pipeline/auxiliary/number.ts]: Solved.

File: src/pipeline/show-nonprinting.ts (hunk 5 above)
Comment (oliverjiu) [reply src/pipeline/show-nonprinting.ts]: Solved.

**Comment by winterhotlatte** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 25
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/cpuabuse/cross-cat/pullRequest?prid=5215125&bid=17100350)

File: src/pipeline/auxiliary/number.ts [hunk 6]
```diff
@@ -1,109 +1,116 @@
...
+/**
+ * Numbers the line.
+ * @param text Text to check
+ * @param countBlank Text to check
+ */
+export function number(text: string, countBlank: boolean): string {
+	// Split the text into lines
+	let splitText: Array<string> = text.split("\n");
+
+	//
```
Comment (yanocv) [inline src/pipeline/auxiliary/number.ts]: What happened here?

**Review by yanocv — CHANGES_REQUESTED** [review]
Oops, found another one. Please fix it.

**Comment by yanocv** [comment]
The requested changes are ok. Good job.

**Comment by winterhotlatte** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 25
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/cpuabuse/cross-cat/pullRequest?prid=5215125&bid=17100350)

File: src/pipeline/auxiliary/number.ts (hunk 6 above)
Comment (oliverjiu) [reply src/pipeline/auxiliary/number.ts]: I might have accidentally deleted the comment.

**Review by yanocv — APPROVED** [review]
Nice.

**Pull request merged**